include/Zycore/Defines.h

```diff
@@ -152,12 +178,54 @@
 #   define ZYAN_PPC64
 #elif defined(__powerpc__)
 #   define ZYAN_PPC
-#elif defined(__riscv) && __riscv_xlen == 64
-#   define ZYAN_RISCV64
+#elif defined(__riscv) || defined(__riscv__)
+#   if __riscv_xlen == 64
+#       define ZYAN_RISCV64
+#   else
+#       define ZYAN_RISCV32
+#   endif
+#elif defined(__arc__)
+#   define ZYAN_ARC
+#elif defined(__s390x__)
+#   define ZYAN_S390
+#elif defined(__sparc__)
+#   define ZYAN_SPARC
+#elif defined(__mips__)
+#   define ZYAN_MIPS
 #else
 #   error "Unsupported architecture detected"
 #endif
 
+/* ============================================================================================== */
+/* Endianness detection                                                                           */
+/* ============================================================================================== */
+
+#define ZYAN_LITTLE_ENDIAN 0
+#define ZYAN_BIG_ENDIAN 1
+
+#if !defined(ZYAN_NO_LIBC) && ZYAN_HAS_INCLUDE(<stdbit.h>)
```

Comment:
One would think that using `__STDC_VERSION__` to check for C23 would be enough but it's not. MSVC support for C23 library features is abysmal and on *nix `-march` doesn't tell you anything about glibc that will be used.